asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9.8-extension-timestamp.adoc

```diff
@@ -0,0 +1,86 @@
+[#vol3_clause_timestamp_versioning]
+====== Timestamp versioning
+
+BICEPS does not provide any means to convey step-changes in a <<vol1_spec_sdpi_p_actor_somds_participant>>'s local time-reference frame (see <<vol1_clause_appendix_c_use_case_stad_non_slew, use case for non-slewing time adjustments>>). 
+
+A <<vol1_spec_sdpi_p_actor_somds_provider>> includes timestamps in many state updates including `pm:AlertConditionState/@DeterminationTime`, `pm:AbstractMetricValue/@DeterminationTime` and `pm:AbstractContextState/@BindingStartTime`. From time-to-time, though rarely in normal operation, a 
+<<vol1_spec_sdpi_p_actor_somds_participant>> may determine that the difference between its time-reference frame and that of the <<acronym_ts_service>> is greater than can be accomodated by smooth adjustments to its clock. This may occur, for example:
+
+* when the <<acronym_ts_service>> is unreachable for a prolonged period, or
+* following hardware failures and/or operator errors in the <<vol1_spec_sdpi_p_actor_somds_provider>> and/or <<acronym_ts_service>>, or
+* after switching to a different and/or backup <<acronym_ts_service>> when the primary <<acronym_ts_service>> becomes unavailable, or
+* when network congestion leads to asymmetrical network transport delays while exchanging messages with the <<acronym_ts_service>>.
+
+In <<ref_rfc_5905>> this is referred to as a step-adjustment or a non-slewing time adjustment. In the absence of step-adjustments, timestamps generated within a <<vol1_spec_sdpi_p_actor_somds_participant>>'s time-reference frame are well-behaved:
+
+* they never decrease,
+* have a well defined relationship to timestamps within the same time-reference frame, and
+* have well defined relationships to peer time sources.
+
+The presence of non-slewing time adjustments creates epochs of consistency: punctuated equilibria where the <<vol1_spec_sdpi_p_actor_somds_participant>>'s timestamps are well-behaved, separated by step-changes. At best, epochs are separated by a constant temporal offset; at worst <<vol1_spec_sdpi_p_actor_somds_participant>>s may have insufficient information to determine the relationship between epochs (e.g., changes at the <<acronym_ts_service>> that do not represent a change in elapsed time to unbiased observers). 
+
+[NOTE]
+====
+R1520 excludes non-slewing adjustments to the <<acronym_ts_service>> by the RESPONSIBLE ORGANIZATION during normal operation. 
```

Comment:
My understanding is R1520 requires provision of a time-service with a 50 ms accuracy. So either the difference between a primary time source and the time reported by the time-service is never greater than 50 ms or the time-service is the authoritative definition of time for SDC participants. Such a time-source could never require a non-slewing adjustment without violating the requirement so I can't see how it could be the source of one for a participant. 

I can't quite see how any problems with the time-service could be resolved. Perhaps by shutting down all SDC devices on the network fixing the clock then restarting everything? But if the time-service is the authoritative definition of time for SDC participants it is always right anyway.